src/quantization/fp32_quantizer.h

```diff
@@ -56,10 +66,16 @@ class FP32Quantizer : public Quantizer<FP32Quantizer<Metric>> {
                     float* dists) const;
 };
 
+template <MetricType Metric>
+FP32Quantizer<Metric>::FP32Quantizer(const nlohmann::json& quantization_obj,
+                                     const IndexCommonParam& common_param)
+    : Quantizer<FP32Quantizer<Metric>>(common_param.dim_) {
+    this->code_size_ = common_param.dim_ * sizeof(float);
+}
+
 template <MetricType Metric>
 FP32Quantizer<Metric>::FP32Quantizer(int dim) : Quantizer<FP32Quantizer<Metric>>(dim) {
-    // align 64 bytes (512 bits) to avoid illegal memory access in SIMD
-    this->code_size_ = (dim * sizeof(float) + (1 << 6) - 1) >> 6 << 6;
+    this->code_size_ = dim * sizeof(float);
```

Comment:
use the align logic is one solution to pass the asan test, but the meaning of the codesize will be misunderstanding. actually, fix simd operators to avoid memory access out of bound may be a better solution. Currently we only test the dim % 8 == 0 cases to avoid asan warnings

Reply:
OK